chap-cheri-riscv.tex

```diff
@@ -502,34 +484,22 @@ \subsubsection{Capability Exception Reporting}
 general-purpose capability register file.
 \end{description}
 
-\subsubsection{Capability Control and Status Registers (CCSRs)}
-\label{subsubsec-ccsrs}
-New per HART \xccsr{} \texttt{XLEN}-bit RISC-V CSRs are defined as per
-Figure~\ref{fig-ccsr} (shown for XLEN=32):
-
-\begin{figure}[!h]
-\begin{center}
-\begin{bytefield}[bitwidth=\textwidth/34]{32}
-  \bitheader[endianness=big]{0,1,2,31} \\
-  \bitbox{31}{\textbf{WIRI}}
-  \bitbox{1}{\texttt{e}}
-\end{bytefield}
-\caption{\xccsr{} register format; WIRI bits are Write Ignore Read Ignore.}
-\label{fig-ccsr}
-\end{center}
-\end{figure}
+\subsubsection{CHERI Extension Control}
 
-\pmnote{WIRI has been removed from the RISC-V specification.  The spec
-  now typically uses WPRI (Write-Preserve, Read-Ignore) for such
-  undefined bits (e.g. in \texttt{mstatus}).  Both WARL (Write-Any,
-  Read-Legal) and WLRL (Write-Legal, Read-Legal) seem unsuitable here.
-  The RISC-V spec unsatisfyingly does not specify what happens if
-  software doesn't preserve values for WPRI, but then this spec
-  doesn't need to either.}
+For processors supporting the \texttt{Smstateen} extension, bit 2 of
```

Comment:
I'm not quite clear on what the difference is between `menvcfg` (privspec 1.12) and [`mstateen`](https://github.com/riscv/riscv-state-enable/blob/main/content.adoc)? Both seem to be used to define access to extensions. I think we definitely need the `mstateen` to gate access to CHERI registers and SCRs, but do we also need a bit in menvcfg to limit access to the instructions?

Reply:
I guess the only instructions we have that don't access the registers are CRRL/CRAM and there shouldn't be much harm in always providing them (unless we want to have a mode where CHERI is  not available at all and all non-baseline ISA encodings trap).

Reply:
I was not aware of `menvcfg` when I wrote this text.  Certainly it might be that we might want that either in addition or possibly in place of just `mstateen`.  In particular, CHERI is not really just a side extension with some state like floating-point or crypto acceleration but instead makes non-trivial changes to how the base ISA is used.

Reply:
In theory if you don't touch CHERI state you shouldn't know it's there, and that's true other than for memory accesses that wrap the address space where we will erroneously trap with a bounds fault even if DDC/PCC covers the whole address space

Reply:
I'm a bit concerned that bit 2 will conflict with the next extension that gets added to RISC-V. Should we pick something further the middle (e.g. 0xc/0x1c for CHERI?).

Reply:
So if they are added in order then bit 31 is safest for now since it won't conflict with anything until all bits are taken?

Reply:
I guess so.  We probably want some sort of note that the bit is subject to change when/if CHERI is ratified upstream.  We could perhaps use bit 0 (defined for "custom state") for now until CHERI is ratified?  I also need to go read the `senvcfg` spec to see how that applies here and if we want both bits or only one of them.

Reply:
I have read a bit more of some of the specs using `Xenvcfg` btw, and I do think that actually it is a better fit.  Given how this bit is currently defined, once CHERI is disabled you can't access the upper bits of registers that have been extended to capabilities or entirely new capability registers (such as DDC).  As such, I don't think it really makes sense to think of this in terms of enabling/disabling additional state.  Probably it is more about controlling the ability to execute new CHERI instructions for which `Xenvcfg` as used by the `Zcmo*` extensions is a better fit.  There is no reserved bit for local state, so we'd have to just pick a bit that fits in the lower 32 for now (probably bit 8 above `CBZE`) and just add a note that it is subject to change based on ratification.  Existing extension proposals include similar language about the bits possibly moving.